routes/auth.go

```diff
@@ -1,14 +1,67 @@
 package routes
 
 import (
+	"crypto/ecdsa"
 	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
 	"net/http"
 	"regexp"
 	"rest-api/logging"
+	"rest-api/models"
+	"rest-api/util"
+	"time"
 
-	"github.com/gin-gonic/gin"
+	zerolog "github.com/rs/zerolog/log"
 )
 
+const DevPrivK = `-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEINbeLv3X0Lwy03t4n8DOo8V5tOFcMW00kjiopeUi4xAnoAoGCCqGSM49
+AwEHoUQDQgAEteZmZdBkEXr+MruFSHYDs75ArTDxouUDFsC1WEmAkbfh7zUp6ybM
+9a54lQNkHZ2uW7pQ+9JkqTzvYU+pt7uAMg==
+-----END EC PRIVATE KEY-----
+`
+
+const DevPubK = `-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEteZmZdBkEXr+MruFSHYDs75ArTDx
+ouUDFsC1WEmAkbfh7zUp6ybM9a54lQNkHZ2uW7pQ+9JkqTzvYU+pt7uAMg==
+-----END PUBLIC KEY-----
+`
+
+const refreshTokenDuration = time.Hour * 24 * 7
+const accessTokenDuration = time.Minute * 10
+
+var tokenPrivateKey *ecdsa.PrivateKey
+var tokenPublicKey *ecdsa.PublicKey // nolint:unused
+
+func SetupTokenSecrets(mode string, privateKey string, publicKey string) {
+	if privateKey == "" || publicKey == "" {
+		if mode == gin.ReleaseMode {
+			zerolog.Fatal().Msg("could not find jwt-private.pem and/or jwt-public.pem")
+		} else {
+			privateKey = DevPrivK
+			publicKey = DevPubK
+			zerolog.Warn().Msg("using the dev keys for JWTs")
```

Comment:
I would prefer a different way of using the development key.
Do not hard-code any keys, please! We should instead terminate the program and add the development key as fallback in the `Dockerfile`.

Reply:
In production we do [terminate](https://github.com/helpwave/rest-api/pull/112/files#diff-0aa13a94b4c3fde198cb2fc3315c115bc8c5c1d09a0062c73ece4d9bb67c4df8R44) in that case. For development I see no issue in hardcoded keys. The Dockerfile however is not the right place for this, in case people want to develop on their own system. We may want to think about the ability to generate keypairs in the rest-api in this case, but that would seem like an awful lot of code used in development setups only.